crates/apollo-parser/src/parser/grammar/value.rs

```diff
@@ -50,7 +50,22 @@ pub(crate) fn value(p: &mut Parser) {
         }
         Some(T!['[']) => list_value(p),
         Some(T!['{']) => object_value(p),
-        _ => p.err("expected a valid Value"),
+        Some(token) => {
+            let error_message = format!("expected a valid Value got {token:?}");
+            if pop_on_error {
+                p.err_and_pop(&error_message);
+            } else {
+                p.err(&error_message);
+            }
+        }
```

Comment:
i am wondering if when we have an unclosed string `Value`, we should be getting two different errors: a lexical _and_ a syntactic error. That is: "you have an unclosed string" and "this is an unsupported value".